pootle/apps/pootle_store/views.py

```diff
@@ -138,10 +138,7 @@ def get_step_query(request, units_queryset):
                     checks = request.GET['checks'].split(',')
                 elif 'category' in request.GET:
                     category_name = request.GET['category']
-                    try:
-                        category = get_category_id(category_name)
-                    except KeyError:
-                        raise Http404
+                    category = get_category_id(category_name)
```

Comment:
@ta2-1 atm there is almost *no validation whatsoever*

this is very much something im addressing in later PRs